# PART -1 PHYTON BASICS (VARIABLES)

1. Print Your Name with your Father name and Date of birth using suitable escape sequence charactor

In [10]:
name = "Dawood Shaikh"
fatherName = "Shaikh Azeem Uddin"
dob = "02-01-2005"
print("My name is " , name ,"\nMy father's name is ",fatherName,"\nMy DOB is ",dob);

My name is  Dawood Shaikh 
My father's name is  Shaikh Azeem Uddin 
My DOB is  02-01-2005


2. Write your small bio using variables and print it using print function

In [16]:
name = "Dawood Shaikh"
fatherName = "Shaikh Azeem Uddin"
dob = "02-01-2005"
education = "Undergraduate"
university = "SMI University"
fieldOfStudy= "Cyber Security"
experienceIn = "C, Java, Typescript and Python"
hobbies = "reading books and listening to intellectual influencers"
print(f"Hello everyone! my name is {name} and my Father's name is {fatherName}.\nI was born on {dob}.\nI am an {education} student at {university}.")
print(f"My field of study is {fieldOfStudy}, I have experience in using {experienceIn} languages.\nMy hobbies are {hobbies}.")

Hello everyone! my name is Dawood Shaikh and my Father's name is Shaikh Azeem Uddin.
I was born on 02-01-2005.
I am an Undergraduate student at SMI University.
My field of study is Cyber Security, I have experience in using C, Java, Typescript and Python languages.
My hobbies are reading books and listening to intellectual influencers.


3. Write a program in which use all the operators we can use in Python

In [42]:
print("\t--- Arithmetic Operators---\n")
x = 15;y = 10
print(f"Sum of x and y is {x+y}");print(f"Subtraction of x and y is {x-y}");
print(f"Multiplication of x and y is {x*y}");print(f"Division of x and y is {x/y}");
print(f"Power of x tends to y is {x**y}");print(f"Mode of x and y is {x%y}");

print("\n")

print("\t--- Comparison Operators---\n")
x = 15;y = 10
print(f"Is x greater than y?- {x>y}");print(f"Is x less than y?- {x<y}");
print(f"Is x equals to y?- {x==y}");print(f"Is y is greater than equal to x?- {y>=x}");
print(f"Is y is less than equal to x?- {y<=x}");print(f"Is y not equals to x?- {y!=x}");

print("\n")

print("\t--- Assignment Operators---\n")
x = 15;y = 10
print(f"Current value of x is {x}")
x = x+2 # x +=2
print(f"After adding 2 in x: {x}")

print(f"Current value of y is {y}")
y *=2;
print(f"After multiplying it with 2: {y}")

print("\n")

print("\t--- Logical Operators---\n")
name = "Dawood Shaikh"
password = "123"
if( name == "Dawood Shaikh" and password == "123"):
    print(f"Welcome Mr.{name}")
else:
    print(f"Invalid username and Password")

if(name == "Azeem Shaikh" or password == "12345"):
    print(f"Welcome Mr.{name}")
else:
    print(f"Invalid username and Password")

if(not name == "HackerX" and password == "123"):
    print(f"Welcome Mr.{name}")
else:
    print(f"Invalid username and Password")


	--- Arithmetic Operators---

Sum of x and y is 25
Subtraction of x and y is 5
Multiplication of x and y is 150
Division of x and y is 1.5
Power of x tends to y is 576650390625
Mode of x and y is 5


	--- Comparison Operators---

Is x greater than y?- True
Is x less than y?- False
Is x equals to y?- False
Is y is greater than equal to x?- False
Is y is less than equal to x?- True
Is y not equals to x?- True


	--- Assignment Operators---

Current value of x is 15
After adding 2 in x: 17
Current value of y is 10
After multiplying it with 2: 20


	--- Logical Operators---

Welcome Mr.Dawood Shaikh
Invalid username and Password
Welcome Mr.Dawood Shaikh


4. Completes the following steps of small task:
    - Mention Marks of English , Islamiat and Maths out of 100 in 3 different variables
    - Mention Variable of Total Marks and assign 300 to it
    - Calculate Percentage

In [51]:
englishMarks = 70; islamiatMarks = 84; mathsMarks = 100;
totalMarks = 300; obtainedMarks = englishMarks + islamiatMarks + mathsMarks;
yourPercentage = ((obtainedMarks*100)/totalMarks)
print(f"Marks of Islamiat : {islamiatMarks}.\nMarks of English : {englishMarks}.\nMarks of Maths : {mathsMarks}");
print(f"Total Marks: {totalMarks}.\nObtained Marks:{obtainedMarks}");
print(f"Your percentage is {yourPercentage:.2f}");

Marks of Islamiat : 84.
Marks of English : 70.
Marks of Maths : 100
Total Marks: 300.
Obtained Marks:254
Your percentage is 84.67


# PART -2 PHYTON BASICS (CONDITIONAL STATEMENTS)

1. A company decided to give bonus of 5% to employee if his/her year of service is more than 5 years. Ask user for their salary and year of service and print the net bonus amount.

In [56]:
salary = float(input("Enter you salary?: "))
yearsOfService= int(input("Enter your years of service?:"))
if(yearsOfService > 5):
    print(f"Congrats! 5% bonus is added in your salary.")
    print(f"Now you salary is {salary+(salary * 0.05)}")
else:
    print(f"Sorry bonus is not added in your salary");

Enter you salary?:  10000
Enter your years of service?: 6


Congrats! 5% bonus is added in your salary.
Now you salary is 10500.0


2. Write a program to check whether a person is eligible for voting or not. (accept age from user) if age is greater than 17 eligible otherwise not eligible

In [59]:
userAge = int(input("Enter your age?: "));
if(userAge >= 18):
    print("You are eligible for voting");
else:
    print("You are not eligible for voting");

Enter your age?:  18


You are eligible for voting


3. Write a program to check whether a number entered by user is even or odd.

In [61]:
num = int(input("Enter any number?: "));
if(num%2==0):
    print("Number is even")
else:
    print("Number is odd")

Enter any number?:  99


Number is odd


4. Write a program to check whether a number is divisible by 7 or not. Show Answer

In [63]:
num = int(input("Enter any number?: "));
if(num%7==0):
    print("Number is divisible by 7");
else:
    print("Number is not divisible by 7");
    

Enter any number?:  49


Number is divisible by 7


5. Write a program to display "Hello" if a number entered by user is a multiple of five , otherwise print "Bye".

In [66]:
num = int(input("Enter any number?: "));
if(num%5==0):
    print("Hello");
else:
    print("Bye");
    

Enter any number?:  885


Hello


6. Write a program to display the last digit of a number.

In [71]:
num = 849223409;
num %=10;
print(f"Last digit of a given number is {num}")

Last digit of a given number is 9


7. Take values of length and breadth of a rectangle from user and print if it is square or rectangle.

In [75]:
length = float(input("Enter value for length: "));
breadth = float(input("Enter value for breadth: "));
if(length == breadth):
    print("It is a square");
else:
    print("It is a rectangle");

Enter value for length:  5.5
Enter value for breadth:  5.5


It is a square


8. Take two int values from user and print greatest among them.

In [78]:
num1 = int(input("Enter first number: "));
num2 = int(input("Enter second number: "));
if(num1>num2):
    print(f"The greatest number is {num1}");
else:
    print(f"The greatest number is {num2}");

Enter first number:  25
Enter second number:  29


The greatest number is 29


9. A shop will give discount of 10% if the cost of purchased quantity is more than 1000. Ask user for quantity Suppose, one unit will cost 100. Judge and print total cost for user.

In [82]:
purchasedQauntity =float (input("Enter purchasing amount: "));
if(purchasedQauntity > 1000):
    print(f"Congrats! you got the 10% discount on your Rs.{purchasedQauntity} purchase");
    print(f"After deduction, now you have to pay only Rs.{purchasedQauntity-(purchasedQauntity*0.10)}");
else:
    print(f"Sorry you didn't get any discount, you have to pay exact Rs.{purchasedQauntity}");

Enter purchasing amount:  10000


Congrats! you got the 10% discount on your Rs.10000.0 purchase
After deduction, now you have to pay only Rs.9000.0


10. A school has following rules for grading system:

a. Below 25 - F

b. 25 to 45 - E

c. 45 to 50 - D

d. 50 to 60 - C

e. 60 to 80 - B

f. Above 80 - A

Ask user to enter marks and print the corresponding grade.

In [95]:
marks = float(input("Enter your marks: "));
if(marks >=80):
    print("Your grade is 'A'");
elif(marks>=60):
    print("Your grade is 'B'");
elif(marks>=50):
    print("Your grade is 'C'");
elif(marks>=45):
    print("Your grade is 'D'");
elif(marks>=25):
    print("Your grade is 'E'");
elif(marks<25):
    print("Your grade is 'F'");
else:
    print("Invalid grade");




Enter your marks:  80


Your grade is 'A'


11. A student will not be allowed to sit in exam if his/her attendence is less than 75%.

Take following input from user

    - Number of classes held

    - Number of classes attended.

And print

    - percentage of class attended

    - Is student is allowed to sit in exam or not.

In [100]:
noOfClasses = int(input("Enter totol number of classes: "));
noOfClassesAttended = int(input("Enter number of classes you have attended so far: "));
percentageOfAttendedClasses = (noOfClassesAttended * 100)/noOfClasses
if(percentageOfAttendedClasses < 75 ):
    print(f"Your Attendance percentage is {percentageOfAttendedClasses:.2f}.\nYou are not allowed to sit in the exam");
else:
    print(f"Your Attendance percentage is {percentageOfAttendedClasses:.2f}.\nYou are allowed to sit in the exam");

Enter totol number of classes:  14
Enter number of classes you have attended so far:  8


Your Attendance percentage is 57.14.
 You are not allowed to sit in the exam


12. Modify the above question to allow student to sit if he/she has medical cause. Ask user if he/she has medical cause or not ( 'Y' or 'N' ) and print accordingly.

In [114]:
noOfClasses = int(input("Enter totol number of classes: "));
noOfClassesAttended = int(input("Enter number of classes you have attended so far: "));
medicalCause = input("( 'Y' or 'N' ) if you have medical cause");

percentageOfAttendedClasses = (noOfClassesAttended * 100)/noOfClasses
if(percentageOfAttendedClasses < 75 and (medicalCause == 'n' or medicalCause == 'N')):
    print(f"Your Attendance percentage is {percentageOfAttendedClasses:.2f}.\nYou don't have any medical cause.\nSo you are not allowed to sit in the exam.");
elif((medicalCause == 'y' or medicalCause == 'Y') and percentageOfAttendedClasses < 75 ):
    print(f"Your Attendance percentage is {percentageOfAttendedClasses:.2f}.\nBut you have a medical cause.\nThat's why you are allowed to sit in the exam.");
else:
    print(f"Your Attendance percentage is {percentageOfAttendedClasses:.2f}.\nYou are allowed to sit in the exam");

Enter totol number of classes:  14
Enter number of classes you have attended so far:  10
( 'Y' or 'N' ) if you have medical cause Y


Your Attendance percentage is 71.43.
But you have a medical cause.
That's why you are allowed to sit in the exam.


13. Write a program to check if a year is leap year or not.
If a year is divisible by 4 then it is leap year but if the year is century year like 2000, 1900, 2100 then it must be divisible by 400.

In [130]:
year = int(input('Enter year you want to check whether it is leap or not: '));
if(year % 4==0 and (not year%100 ==0 or year%400 ==0)):
    print(f"{year} is a leap year");
else:
    print(f"{year} is not a leap year");
    

Enter year you want to check whether it is leap or not:  2024


2024 is a leap year


14. Ask user to enter age, gender ( M or F ), marital status ( Y or N ) and then using following rules print their place of service.

if employee is female, then she will work only in urban areas.

if employee is a male and age is in between 20 to 40 then he may work in anywhere

if employee is male and age is in between 40 t0 60 then he will work in urban areas only.

And any other input of age should print "ERROR"

In [135]:
age = int(input("Enter your age: "));
gender = input("Enter your gender (M or F): ");
maritalStatus = input("Enter your marital status (Y or N): ")
if(gender == "f" or gender == "F" ):
    print("You are female so you will only work in urban areas");
elif((gender == "m" or gender == "M") and (age>=20 and age<=40)):
    print("You are male and your age is between 20 to 40, so you can work in anywhere");
elif((gender == "m" or gender == "M") and (age>=40 and age<=60)):
    print("You are male and your age is between 40 to 60, so you can work only in urban areas");
else:
    print("ERROR");

Enter your age:  25
Enter your gender (M or F):  m
Enter your marital status (Y or N):  y


You are male and your age is between 20 to 40, so you can work in anywhere


15. Write a program to calculate the electricity bill (accept number of unit from user) according to the following criteria : Unit Price
uptp 100 units no charge Next 200 units Rs 5 per unit After 200 units Rs 10 per unit (For example if input unit is 350 than total bill amount is Rs.3500 (For example if input unit is 97 than total bill amount is Rs.0 (For example if input unit is 150 than total bill amount is Rs.750

In [140]:
noOfUnits = float(input("Enter no.of consumed units: "));
if(noOfUnits < 100):
    print("Total bill amount is Rs.0");
elif(noOfUnits <= 200):
    print(f"Total bill amount is Rs.{noOfUnits*5}");
elif(noOfUnits > 200):
    print(f"Total bill amount is Rs.{noOfUnits*10}");

Enter no.of consumed units:  205


Total bill amount is Rs.2050.0


16. Take input of age of 3 people by user and determine oldest and youngest among them.

In [148]:
person_1 = int(input("Enter the age of first person: "));
person_2 = int(input("Enter the age of second person: "));
person_3 = int(input("Enter the age of third person: "));
if(person_1 > person_2 and person_1>person_3):
    print(f"First person is the oldest one!, whose age is {person_1}.");
elif(person_2 > person_1 and person_2>person_3):
    print(f"Second person is the oldest one!, whose age is {person_2}");
elif(person_3 > person_1 and person_3>person_2):
    print(f"Third person is the oldest one!, whose age is {person_3}");
elif(person_1 == person_2 and person_2>person_3):
    print(f"First and Second both are the oldest one!, whose age is {person_3}");
elif(person_1 == person_3 and person_1>person_2):
    print(f"First and Third both are the oldest one!, whose age is {person_3}");
elif(person_2 == person_3 and person_2>person_1):
    print(f"Second and Third both are the oldest one!, whose age is {person_3}");

if(person_1 < person_2 and person_1<person_3):
    print(f"First person is the youngest one!, whose age is {person_1}.");
elif(person_2 < person_1 and person_2<person_3):
    print(f"Second person is the yougest one!, whose age is {person_2}");
elif(person_3 < person_1 and person_3< person_2):
    print(f"Third person is the youngest one!, whose age is {person_3}");
elif(person_1 == person_2 and person_2<person_3):
    print(f"First and Second both are the youngest one!, whose age is {person_3}");
elif(person_1 == person_3 and person_1<person_2):
    print(f"First and Third both are the youngest one!, whose age is {person_3}");
elif(person_2 == person_3 and person_2<person_1):
    print(f"Second and Third both are the youngest one!, whose age is {person_3}");

if(person_1==person_2 and person_2 == person_3):
    print(f"All persons have the same age that is {person_1}")

Enter the age of first person:  16
Enter the age of second person:  14
Enter the age of third person:  16


First and Third both are the oldest one!, whose age is 16
Second person is the yougest one!, whose age is 14
